openapi.yaml

```diff
@@ -139,54 +139,72 @@ info:
     
     openEO-based back-ends are usually hosted on a different domain / host than the client that is requesting data from the back-end. Therefore most requests to the back-end are blocked by all modern browsers. This leads to the problem that the JavaScript library and any browser-based application can't access back-ends. Therefore, all back-end providers SHOULD support CORS to enable browser-based applications to access back-ends. [CORS is a recommendation of the W3C organization](https://www.w3.org/TR/cors/). The following chapters will explain how back-end providers can implement CORS support.
     
-    ## OPTIONS method
+    ## CORS headers
     
-    All endpoints must respond to the `OPTIONS` HTTP method. This is a response for the preflight requests made by the browsers. It needs to respond with a status code of `204` and send the HTTP headers shown in the table below. No body needs to be provided.
+    The following headers MUST be included with every response:
     
     | Name                             | Description                                                  | Example |
     | -------------------------------- | ------------------------------------------------------------ | ------- |
-    | Access-Control-Allow-Origin      | Allowed origin for the request, including protocol, host and port. It is RECOMMENDED to return the value of the request's origin header. If no `Origin` is sent to the back-end CORS headers SHOULD NOT be sent at all. | `http://client.isp.com:80` |
-    | Access-Control-Allow-Credentials | If authorization is implemented by the back-end the value MUST be `true`. | `true` |
-    | Access-Control-Allow-Headers     | Comma-separated list of HTTP headers allowed to be send. MUST contain at least `Authorization` if authorization is implemented by the back-end. | ` Authorization, Content-Type` |
-    | Access-Control-Allow-Methods     | Comma-separated list of HTTP methods allowed to be requested. Back-ends MUST list all implemented HTTP methods for the endpoint here. | `OPTIONS, GET, POST, PATCH, PUT, DELETE` |
+    | Access-Control-Allow-Origin      | Allowed origin for the request, including protocol, host and port or `*` for all origins. It is RECOMMENDED to return the value `*` to allow requests from browser-based implementations such as the Web Editor. | `*` |
     | Access-Control-Expose-Headers    | Some endpoints send non-safelisted HTTP response headers such as `OpenEO-Identifier` and `OpenEO-Costs`. All headers except `Cache-Control`, `Content-Language`, `Content-Type`, `Expires`, `Last-Modified` and `Pragma` must be listed in this header. Currently, the openEO API requires at least the following headers to be listed: `Location, OpenEO-Identifier, OpenEO-Costs, Link`. | `Location, OpenEO-Identifier, OpenEO-Costs, Link` |
```

Comment:
> Some endpoints send non-safelisted HTTP response headers such as `OpenEO-Identifier` and `OpenEO-Costs`.

I know it's the technical term, but "endpoints send non-safelisted" sounds like we are doing insecure things. I would change the wording a bit to e.g.

> Some endpoints send important information through custom HTTP response headers such as `OpenEO-Identifier` and `OpenEO-Costs`.


I also think we can drop this sentence:
> All headers except `Cache-Control`, `Content-Language`, `Content-Type`, `Expires`, `Last-Modified` and `Pragma` must be listed in this header.

It's a general CORS implementation detail that is not very relevant here. 
Especially the "all headers" can be confusing: is it "all possible HTTP headers under the sun", "all relevant headers", ...? 
I think the example is good enough to make it clear what a backend should do.

Reply:
Hmm, I don't fully agree with this.

> through custom HTTP response headers

Location for example is not really a "custom" HTTP header for me.

Also, I'd personally appreciate if somebody tells me which headers not to list.

Anyway, I tried to improve the situation a bit, e.g. by using white-listed instead of non-safelisted and explaining that "All headers that should be made available to browser-based clients MUST be listed in this header."

Reply:
ok, good point.

However, I think this sentence is now bit weird:
> In addition to the white-listed HTTP headers ..., some endpoints need to send additional HTTP ...

I would keep the "some endpoints" statement separate from the white list (which is a global thing).

suggestion:

>  Some endpoints responses include HTTP headers such as `OpenEO-Identifier` (e.g. containing id of a newly created job) and `Location` (for forwarding). To make these headers available to browser based clients, they MUST be white-listed through the `Access-Control-Expose-Headers` response header. (The following headers are white-listed by default and should not be included: `Cache-Control`, `Content-Language`, `Content-Length`, `Content-Type`, `Expires`, `Last-Modified` and `Pragma`.) This version of the openEO API requires at least the following headers to be listed: `Link`, `Location`, `OpenEO-Costs` and `OpenEO-Identifier`.

Note: 
- from https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Expose-Headers I learned that `Content-Length` is also whitelisted by default and I added it to the list. Feel free to revert if that is wrong
- I also changed `Currently,` to `This version of the api` to be a bit more explicit version-wise

Reply:
Good catch on Content-Length! My source didn't list it. Interestingly, it is missing in several sources as it was added later: https://github.com/whatwg/fetch/pull/626

@soxofaan I made the suggested changes, but tried to be a bit more compact. Is this ready to be merged now?

Reply:
yes, looks lot better now